reproject/interpolation/core_celestial.py

```diff
@@ -48,12 +51,20 @@ def _reproject_celestial(array, wcs_in, wcs_out, shape_out, order=1):
     # remainder of the array. We then operate on the view, but this will change
     # the original array with the correct shape.
 
-    array_new = np.zeros(shape_out)
+    if array_out is not None:
+        if tuple(array_out.shape) != tuple(shape_out):
```

Comment:
Just out of curiosity, why the ``tuple``s? The first one should be unnecessary at the very least?